acl/authorizer.go

```diff
@@ -161,6 +162,280 @@ type Authorizer interface {
 
 	// Embedded Interface for Consul Enterprise specific ACL enforcement
 	enterpriseAuthorizer
+
+	// ToAllowAuthorizer is needed until we can use ResolveResult in all the places this interface is used.
+	ToAllowAuthorizer() AllowAuthorizer
```

Comment:
Does this need to be a method? Do any of the implementation of this interface do more than `&AllowAuthorizer{a}`?

If they are all the same there may be a benefit towards just having `func acl.NewAllowAuthorizer(Authorizer)` instead

Reply:
This is a transitional method; in upcoming PRs more functionality will be loaded onto it (in particular capturing authorization source information), but eventually I'd like to move a bunch of stuff into the ACL directory and hopefully remove this entirely.